### PR TITLE
Allow passing endpoint alias with the -e/--endpoint flag and GSCTL_ENDPOINT env variable

### PR DESCRIPTION
**Pull request by marians**
Fixes https://github.com/giantswarm/gsctl/issues/229

This allows to use endpoint aliases instead of endpoint URLs with the `-e` or `--endpoint` flag in all commands, and with the `GSCTL_ENDPOINT` environment variable.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -206,22 +206,29 @@ func (c *configStruct) SelectEndpoint(endpointAliasOrURL string) error {
...
+			return ep
+		}
+
+		ep := normalizeEndpoint(overridingEndpointAliasOrURL)
 		return ep
 	}
 
+	// use environment variable
 	envEndpoint := os.Getenv("GSCTL_ENDPOINT")
 	if envEndpoint != "" {
```
[the remark this answers is not in the document]
Comment (marians) [reply config/config.go]: Valid point!

[the remark this answers is not in the document]
Comment (marians) [reply config/config.go]: Modified.

[the remark this answers is not in the document]
Comment (pipo02mix) [reply config/config.go]: Nitpick but what about avoiding duplicate with a new method or getting the endpoint first to the logic?
```
if overridingEndpointAliasOrURL == "" {
    overridingEndpointAliasOrURL = os.Getenv("GSCTL_ENDPOINT")
}

if overridingEndpointAliasOrURL != "" {
    // check if overridingEndpointAliasOrURL is an alias
    if c.HasEndpointAlias(overridingEndpointAliasOrURL) {
        ep, _ := c.EndpointByAlias(overridingEndpointAliasOrURL)
        return ep
    }

    ep := normalizeEndpoint(overridingEndpointAliasOrURL)
    return ep
 }
```

[the remark this answers is not in the document]
Comment (marians) [reply config/config.go]: Got your point. Changed.

**Comment by marians** [comment]
@pipo02mix Please re-check. Thanks!

**Comment by marians** [comment]
@pipo02mix Any chance to get a final review?

**Review by pipo02mix — APPROVED** [review]
LGTMIIWGY

**Pull request merged**